eoscompanion/routes.py

```diff
@@ -171,6 +174,102 @@ def companion_app_server_feed_route(server, msg, path, query, *args):
     feed = dummy_feed()
     json_response(msg, feed)
 
+
+_SUFFIX_CONTENT_TYPES = {
+    '.css': 'text/css',
+    '.js': 'application/javascript'
+}
+
+
+@require_query_string_param('deviceUUID')
+@require_query_string_param('uri')
+def companion_app_server_resource_route(server, msg, path, query, *args):
+    '''Fetch an internal resource from a rewritten link on the page.
+
+    This route is for fetching internal resources not tied to any particular
+    application ID or content shard. For instance, we might embed a link
+    to a CSS or JS file that we want to serve up to the client.
+
+    The querystring param "path" indicates the URI encoded path to the
+    internal resource (it may be a GResource or a file).
+    '''
+    del args
+
+    resource_uri = Soup.URI.decode(query['uri'])
+    resource_file = Gio.File.new_for_uri(resource_uri)
+
+    resource_suffix = os.path.splitext(resource_uri)[1]
+    return_content_type = _SUFFIX_CONTENT_TYPES.get(resource_suffix, None)
+
+    if return_content_type is None:
+        json_response(msg, {
+            'status': 'error',
+            'error': {
+                'domain': GLib.quark_to_string(EosCompanionAppService.error_quark()),
+                'code': EosCompanionAppService.Error.FAILED,
+                'detail': {
+                    'server_error': (
+                        'Don\'t know content type for suffix, {}'.format(resource_suffix)
+                    )
+                }
+            }
+        })
+        return
+
+    def _on_read_input_stream(_, result):
+        '''Callback for when we finish reading the input stream.'''
+        try:
+            content_bytes = EosCompanionAppService.finish_load_all_in_stream_to_bytes(result)
+        except GLib.Error as error:
+            json_response(msg, {
+                'status': 'error',
+                'error': {
```

Comment:
Any reason not to use the `serialize_error_as_json_object()` utility here?

Reply:
Omission on my part :)